### PR TITLE
MPDX-7935 add mailing address

### PR DESCRIPTION
**Pull request by dr-bizz**
## Description
In this PR, I added a model for adding a mailing address, cleaned up the add button, and made some improvements to `AddAddressModal`, which I then did on `EditContactAddressModal`. I've also added tests for editing, deleting, and adding an address.

I also cleaned up the FixMailingAddress component, switched the modal state props to separate state values, and refactored the modal open/close handling.

## Checklist:

- [ ] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [ ] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [ ] I have requested a review from another person on the project

### CODE REVIEW

**Comment by caleballdrin** [comment]
@dr-bizz The code looks good! Now that you have the add address button you can remove the "MPDX" text. Could you also align the button with the address boxes?
Screenshot 2024-06-13 at 5 02 53 PM

**Comment by caleballdrin** [comment]
I also noticed that Tnt Imported addresses have a lock next to them. They should show an edit pen icon because they are editable.

**Comment by dr-bizz** [comment]
> I also noticed that Tnt Imported addresses have a lock next to them. They should show an edit pen icon because they are editable.

I don't think they should be editable, as they weren't before on the old MPDX. I think this is due to ensuring TNT is the only place people can update their info if it's imported from TNT. 
Only is the source is MPDX, can it be edited on MPDX.

**Comment by caleballdrin** [comment]
> I don't think they should be editable, as they weren't before on the old MPDX. I think this is due to ensuring TNT is the only place people can update their info if it's imported from TNT. Only is the source is MPDX, can it be edited on MPDX.

@dr-bizz They are editable in the old MPDX, it just also incorrectly shows a lock. There is no automatic update from Tnt like there is for Google and Siebel. Tnt import is more similar to a csv import where it is a one-time upload of data and then users will want to be able to edit that data in MPDX rather than going back to the csv/tnt file to re-import.

**Comment by dr-bizz** [comment]
> > I don't think they should be editable, as they weren't before on the old MPDX. I think this is due to ensuring TNT is the only place people can update their info if it's imported from TNT. Only is the source is MPDX, can it be edited on MPDX.
> 
> @dr-bizz Tnt addresses **are** editable in the old MPDX, it just also incorrectly shows a lock. There is no automatic update from Tnt like there is for Google and Siebel. Tnt import is more similar to a csv import where it is a one-time upload of data and then users will want to be able to edit that data in MPDX rather than going back to the csv/tnt file to re-import.

@caleballdrin Fixed

**Comment by caleballdrin** [comment]
@dr-bizz I'm pretty sure Tnt is different than DataServer. If you look at Pedros account you can see my contact was imported from Tnt.

**Review by caleballdrin — APPROVED** [review]
Nice work!

File: src/components/Tool/FixMailingAddresses/Contact.tsx [hunk 1]
```diff
@@ -233,7 +235,9 @@ const Contact: React.FC<Props> = ({
                       <strong>{t('Source')}: </strong>
                     </Typography>
                   </Hidden>
-                  <Typography display="inline">MPDX</Typography>
+                  <Typography display="inline">
```
Comment (caleballdrin) [inline src/components/Tool/FixMailingAddresses/Contact.tsx:243]: @dr-bizz this should be removed altogether, not changed to appName. "MPDX" is still showing up across from the "Add Address" button.

**Review by caleballdrin — CHANGES_REQUESTED** [review]

**Review by caleballdrin — APPROVED** [review]

**Pull request merged**